core/src/main/java/org/jruby/ext/socket/Option.java

```diff
@@ -166,24 +167,34 @@ private String optionValue() {
         return "";
     }
 
-    @JRubyMethod(meta = true)
-    public IRubyObject rb_int(ThreadContext context, IRubyObject self) {
-        return context.nil;
+    @JRubyMethod(name = "int", required = 4, meta = true)
+    public static IRubyObject rb_int(ThreadContext context, IRubyObject self, IRubyObject[] args) {
+        ProtocolFamily family = SocketUtils.protocolFamilyFromArg(args[0]);
+        SocketLevel level = RubyBasicSocket.levelFromArg(args[1]);
+        SocketOption option = RubyBasicSocket.optionFromArg(args[2]);
+        int intData = RubyNumeric.fix2int(args[3]);
+
+        return new Option(context.getRuntime(), family, level, option, intData);
     }
-    
+
     @JRubyMethod(name = "int")
     public IRubyObject asInt(ThreadContext context) {
         return context.getRuntime().newFixnum((int) intData);
     }
 
-    @JRubyMethod(meta = true)
-    public IRubyObject bool(ThreadContext context, IRubyObject self) {
-        return context.nil;
+    @JRubyMethod(required = 4, meta = true)
+    public static IRubyObject bool(ThreadContext context, IRubyObject self, IRubyObject[] args) {
+        ProtocolFamily family = SocketUtils.protocolFamilyFromArg(args[0]);
+        SocketLevel level = RubyBasicSocket.levelFromArg(args[1]);
+        SocketOption option = RubyBasicSocket.optionFromArg(args[2]);
+        int intData = args[3].isTrue() ? 1 : 0;
```

Comment:
Should it be checked there that it actually a boolean arg and what to do in case if it's not?